### PR TITLE
fix(PE-7090): return records as sorted entries

### PR DESCRIPTION
**Pull request by atticusofsparta**

### CODE REVIEW

File: src/common/records.lua [hunk 1]
```diff
@@ -53,7 +53,21 @@ end
 --- Get all records from the ANT
 ---@return string The encoded JSON representation of all records.
 function records.getRecords()
-	return json.encode(Records)
+	local antRecords = utils.deepCopy(Records)
+	assert(antRecords, "Failed to copy Records")
```
Comment (dtfiedler) [inline src/common/records.lua:57]: what is this protecting against?

File: src/common/records.lua [hunk 2]
```diff
@@ -53,7 +53,21 @@ end
...
+		if a[1] == "@" then
+			return true
+		end
+		if b[1] == "@" then
+			return false
+		end
+		return a[1] < b[1]
+	end)
+
+	return json.encode(recordEntries)
```
Comment (dtfiedler) [inline src/common/records.lua:70]: why do we need to encode here?

File: src/common/records.lua (hunk 1 above)
Comment (atticusofsparta) [reply src/common/records.lua:57]: deepCopy returns table | nil

File: src/common/records.lua (hunk 2 above)
Comment (atticusofsparta) [reply src/common/records.lua]: [30824b5](https://github.com/ar-io/ar-io-ant-process/pull/35/commits/30824b5c9392d826e562c4d833fb7ebed4fe01a7)

File: src/common/records.lua [hunk 3]
```diff
@@ -51,9 +51,23 @@ function records.getRecord(name)
...
+		if a[1] == "@" then
+			return true
+		end
+		if b[1] == "@" then
+			return false
+		end
+		return a[1] < b[1]
+	end)
+
+	return recordEntries
```
Comment (dtfiedler) [inline src/common/records.lua:70]: this should just be a flat array with the record details and key

```
[
{ 
  "record": "@",
   "transactionId: "xxxx",
    "ttlSeconds": 9000
 }
]
```
ANTs can choose how they'll sort this array. by default we can do '@' first and the rest alphabetically.

Comment (atticusofsparta) [reply src/common/records.lua]: [e1f8b75](https://github.com/ar-io/ar-io-ant-process/pull/35/commits/e1f8b755a516bb24c8d746247fffb9355f182b87)

File: src/common/records.lua (hunk 1 above)
Comment (dtfiedler) [reply src/common/records.lua:62]: please add unit tests for this

File: src/common/utils.lua [hunk 4]
```diff
@@ -189,6 +189,16 @@ utils.values = function(t)
...
 
+---@param t table
+---@return table<table>
+utils.entries = function(t)
+	local entries = {}
+	for _, key in ipairs(utils.keys(t)) do
+		table.insert(entries, { key, t[key] })
+	end
+	return entries
+end
```
Comment (dtfiedler) [inline src/common/utils.lua:200]:
```suggestion
```

Comment (atticusofsparta) [reply src/common/utils.lua]: [25dc11a](https://github.com/ar-io/ar-io-ant-process/pull/35/commits/25dc11aaf8ee968c06bf8a7e2aa65046c2e0283d)

File: src/common/records.lua (hunk 1 above)
Comment (atticusofsparta) [reply src/common/records.lua:62]: [5a080c7](https://github.com/ar-io/ar-io-ant-process/pull/35/commits/5a080c72175a6e604b290ae89e5bd53fc3cbbee1)

File: spec/ant_spec.lua [hunk 5]
```diff
@@ -82,6 +82,15 @@ describe("Arweave Name Token", function()
 		assert.are.same(_G.Records["@"].ttlSeconds, 900)
 	end)
 
+	it("gets all records", function()
+		records.setRecord("zed", string.rep("1", 43), 3600)
+		records.setRecord("@", string.rep("1", 43), 3600)
```
Comment (dtfiedler) [inline spec/ant_spec.lua:87]: nit - generally when testing you should avoid "using the tools your testing" - instead, set the records manually in global state using `_G.Records = {}` and _then_ validate the behavior

Comment (atticusofsparta) [reply spec/ant_spec.lua]: good point - https://github.com/ar-io/ar-io-ant-process/pull/35/commits/16220a98103d5b7c913d7c304baeebfa4b66ea9d

**Comment by dtfiedler** [comment]
i'd like to see the changes prepped to support this in the ar-io-sdk. bc we have previously returning maps, we'll want to make the ANT repo backwards compatible to convert maps to arrays. please create a separate ticket and make those changes there.

**Comment by atticusofsparta** [comment]
@dtfiedler https://github.com/ar-io/ar-io-sdk/pull/259

**Pull request merged**